src/index.js

```diff
@@ -139,11 +139,30 @@ async function start() {
     })
   })
 
-  driver.listen(PORT, () => {
+  const server = driver.listen(PORT, () => {
     console.info(
       `🚀 KILT DID resolver driver running on port ${PORT} and connected to ${BLOCKCHAIN_NODE}...`
     )
   })
+
+  // graceful shutdown: stop accepting new requests -> wait for running requests to be completed -> close api connection and exit
+  function shutdown(signal) {
+    console.log(`${signal} signal received: closing HTTP server`)
+    const timeout = setTimeout(() => {
+      console.warn(
+        'timeout for pending requests, force-closing open connections'
```

Comment:
Maybe write how much time has elapsed? Also since we're not using a logger, it might be useful info to have printed.